draft-ietf-quic-transport.md

```diff
@@ -1477,35 +1481,240 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+connection to the new address at a later time.  Failure of path validation
+simply means that the new local address is not usable for this connection and
+should not be fatal to the connection when alternative local addresses are
+available.
+
+A client migrating to a new local address SHOULD use a new connection ID for
+packets sent from that address, see {{migration-linkability}} for further
+discussion.
+
+A client MUST NOT initiate connection migration before the handshake is
+finished and the client has 1-RTT keys.
+
+
+### Responding to Connection Migration
+
+A server may receive a packet from a new client address at any time during the
+connection after the handshake is complete. If the packet is decryptable, the
+client may be either probing from a new address or migrating immediately, as
+described in {{initiating-migration}}.
+
+#### Responding to a Client Probe
+
+Receiving a PATH_CHALLENGE frame from the client indicates that the client is
+probing for server reachability on this new path.  The server sends a
+PATH_RESPONSE to the new client address as per {{migrate-validate}}, but the
+server MUST continue using the current client address for sending all other
+packets.
+
+PATH_CHALLENGE, PATH_RESPONSE, and PADDING frames are "probing" frames, and
+receiving only these frames from a new client address MUST NOT result in the
+server sending any other frames to that address.
+
+#### Responding to an Immediate Migration
+
+Receiving a packet from a new client address indicates that the client may be
+immediately migrating if
+
+* the packet is a "non-probing" packet. That is, it contains a frame other than
+PATH_CHALLENGE, PATH_RESPONSE, or PADDING; and
+
+* the packet carries a packet number that is the largest seen thus far.
+
+Upon receiving such a packet, the server MUST abandon any path validation it is
+performing with other addresses on the expectation that those validations are
+likely to fail.  Abandoning path validation primarily means ceasing subsequent
+transmissions of the PATH_CHALLENGE frame.  An endpoint MUST ignore any
+subsequently received PATH_RESPONSE frames from abandoned addresses.
 
-An endpoint that receives packets that contain a source IP address and port that
-has not yet been used can start sending new packets with those as a destination
-IP address and port.  Packets exchanged between endpoints can then follow the
-new path.
+In response to such a packet, the server MUST start sending subsequent packets
+to this client address and MUST initiate path validation to verify client
+ownership of the unvalidated address.
 
-Due to variations in path latency or packet reordering, packets from different
-source addresses might be reordered.  The packet with the highest packet number
-MUST be used to determine which path to use.  Endpoints also need to be prepared
-to receive packets from an older source address.
+A server MAY skip validation of a client address that it considers to be already
+valid.
```

Comment:
Wow, this invites all sorts of speculation.  I think that you need more explanation here.

Reply:
I understand the desire to leave implementations room to explore the envelope, but I agree this is a bit vague.  I'm assuming this is mostly intended to be read as "same IP, different port" or "return to an IP address the client had used/validated previously" (e.g. cellular, migrated to Wi-Fi, lost Wi-Fi).